### PR TITLE
Add Data_Viz post type

### PR DESCRIPTION
**Pull request by DragonVirus**
- data_viz post type 
- Add ajax function to get ckan datastore
- check for wpckan plugin requirement

### CODE REVIEW

File: post-types/datavizs.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+if (!class_exists('Odm_DataViz_Post_Type')) {
+
+    class Odm_DataViz_Post_Type
+    {
+        public function __construct()
+        {
+          add_action('init', array($this, 'register_post_type'));
+          add_action('save_post', array($this, 'save_post_data'));
+          add_filter('single_template', array($this, 'get_dashboards_template'));
```
Comment (acorbi) [inline post-types/datavizs.php]: @DragonVirus for consitency, I would call this function _get_datavizs_template_

File: post-types/datavizs.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        public function get_dashboards_template($single_template)
```
Comment (acorbi) [inline post-types/datavizs.php]: same as above

File: post-types/datavizs.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            global $post;
+
+            if ($post->post_type == 'dataviz') {
+                if (file_exists(plugin_dir_path(__FILE__).'templates/single-dataviz-'.$post->post_name.'.php')) {
```
Comment (acorbi) [inline post-types/datavizs.php]: @DragonVirus in which scenario would this be of help?

File: post-types/dashboards.php [hunk 4]
```diff
@@ -16,7 +16,11 @@ public function get_dashboards_template($single_template)
             global $post;
 
             if ($post->post_type == 'dashboard') {
-                $single_template = plugin_dir_path(__FILE__).'templates/single-dashboard.php';
+                if (file_exists(plugin_dir_path(__FILE__).'templates/single-dashboard-'.$post->post_name.'.php')) {
```
Comment (acorbi) [inline post-types/dashboards.php:19]: @DragonVirus in which scenario would this be of help?

**Comment by acorbi** [comment]
@DragonVirus FYI: I changed the base of the PR so it gets merged into impl-24 ( the branch for the modularization)

File: post-types/dashboards.php (hunk 4 above)
Comment (DragonVirus) [reply post-types/dashboards.php:19]: custom layout for specific dashboard if needed. 
That's why i added `post-types/templates/single-dashboard-my-overview.php`

File: post-types/datavizs.php (hunk 1 above)
Comment (DragonVirus) [reply post-types/datavizs.php]: Yes ... i will change that.

File: post-types/datavizs.php (hunk 3 above)
Comment (DragonVirus) [reply post-types/datavizs.php]: Ahhh .. Sorry i got confuse this one with dashboard post type. I think we don't need that one.

**Pull request merged**